flight.py

```diff
@@ -0,0 +1,225 @@
+    sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_KEEPCNT, max_fails)
+
+
+def set_keepalive_osx(sock, after_idle_sec=1, interval_sec=3, max_fails=5):
+    """Set TCP keepalive on an open socket.
+
+    sends a keepalive ping once every 3 seconds (interval_sec)
+    """
+    # scraped from /usr/include, not exported by python's socket module
+    TCP_KEEPALIVE = 0x10
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1)
+    sock.setsockopt(socket.IPPROTO_TCP, TCP_KEEPALIVE, interval_sec)
+
+
+class InvalidDirectionException(Exception):
+    """Thrown when specified direction is invalid"""
+    pass
+
+
+class InvalidPriorityException(Exception):
+    """Thrown when specified priority is invalid"""
+    pass
+
+
+def parse_args():
+    """
+
+    Take in Command Line Arguments and give back args to program.
+
+    Returns
+    -------
+    argparse.Namespace
+        Arguments assembled by argparse.
+
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        'MRRDT flight command line interface. Used to fly the drone.')
+    parser.add_argument('-d', '--debug', action="store_true")
+    parser.add_argument('routine', nargs="?", default=None, type=str)
+
+    args = parser.parse_args()
+
+    if not args.debug and not args.routine:
+        logging.error("Please specify routine.")
+        sys.exit(1)
+    return args
+
+
+def create_routes(args, controller):
+    """
+
+    Constructs all routes necessary for network flight
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        Arguments assembled by argparse.
+    controller : flight.drone.drone_controller.DroneController
+        The second parameter.
+
+    Returns
+    -------
+    list
+        Blueprints to be added to Flask app
+
+    """
+    pass
+
+
+def debug_add_task(controller, command, meta):
+    if "priority" in meta:
+        priority = get_priority(meta["priority"])
+    if "direction" in meta:
+        direction = get_direction(meta["direction"])
+
+    if command == "exit":
+        controller.add_exit_task(c.Priorities.HIGH)
+        return False
+    elif command == "land":
+        controller.add_land_task(priority)
+    elif command == "hover":
+        controller.add_hover_task(fc.DEFAULT_ALTITUDE, int(meta["time"]),
+                                  priority)
+    elif command == "takeoff":
+        controller.add_takeoff_task(int(meta["altitude"]))
+    elif command == "move":
+        controller.add_linear_movement_task(direction, int(meta["time"]),
+                                            priority)
+    return True
+
+
+def get_priority(priority):
+    return get_enum(c.Priorities, priority)
+
+
+def get_direction(direction):
+    return get_enum(c.Directions, direction)
+
+
+def get_enum(enum, key):
+    item = key.upper()
+    if hasattr(enum, item):
+        return getattr(enum, item)
+    else:
+        raise InvalidDirectionException(
```

Comment:
The exception is specific to direction, even though it's being used for priority too.